backend/remote-state/s3/backend.go

```diff
@@ -139,6 +139,13 @@ func New() backend.Backend {
 				Description: "The permissions applied when assuming a role.",
 				Default:     "",
 			},
+
+			"workspace_key_prefix": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The prefix applied to the state path inside the bucket",
```

Comment:
Can we mention that it's for non-default workspaces here? Maybe:

    applied to the non-default state path